lib/crypto.js

```diff
@@ -647,9 +647,21 @@ exports.setEngine = function setEngine(id, flags) {
   return binding.setEngine(id, flags);
 };
 
-exports.randomBytes = exports.pseudoRandomBytes = randomBytes;
-
-exports.rng = exports.prng = randomBytes;
+exports.randomBytes = util._deprecateSync(randomBytes,
+    '`crypto.randomBytes(size)` is deprecated. ' +
+    'Please use `crypto.randomBytesSync(size)` instead.');
+exports.randomBytesSync = randomBytes;
+exports.pseudoRandomBytes = util.deprecate(randomBytes,
+    '`crypto.pseudoRandomBytes()` is deprecated. ' +
+    'Please use `crypto.randomBytes()` instead.');
+
+exports.rng = util._deprecateSync(randomBytes,
```

Comment:
Instead of introducing a new deprecation method, you can simply do do the callback check and deprecation message in `node_crypto.cc`. @silverwind, please correct me if I am wrong.